libs/datamodel/core/src/transform/ast_to_dml/db/attributes/id.rs

```diff
@@ -118,10 +121,40 @@ pub(super) fn field<'ast>(
         None => {
             let db_name = primary_key_constraint_name(ast_model, args, "@id", ctx);
 
+            let length = match args.optional_arg("length").map(|length| length.as_int()) {
+                Some(Ok(length)) => Some(length as u32),
+                Some(Err(err)) => {
+                    ctx.push_error(err);
+                    None
+                }
+                None => None,
+            };
+
+            let sort_order = match args.optional_arg("sort").map(|sort| sort.as_constant_literal()) {
+                Some(Ok("Desc")) => Some(SortOrder::Desc),
+                Some(Ok("Asc")) => Some(SortOrder::Asc),
+                Some(Ok(other)) => {
+                    ctx.push_error(args.new_attribute_validation_error(&format!(
+                        "The `sort` argument can only be `Asc` or `Desc` you provided: {}.",
+                        other
+                    )));
+                    None
+                }
+                Some(Err(err)) => {
+                    ctx.push_error(err);
+                    None
+                }
+                None => Some(SortOrder::Asc), //TODO(extended indices) once index types are introduced this needs to distinguish on them. Fulltext has no order for example
```

Comment:
Maybe this code should be in a walker. The walker would know the type of the index and underlying database. The code here would just ask `field_walker.sort_order()`, and get the right answer based on the facts.

Reply:
I removed this for now. This can be done once we do more fine grained validation as well. For now this is about setting the base for the dml work.